plugins/woocommerce/tests/php/src/Internal/StockNotifications/Utilities/StockManagementHelperTests.php

```
@@ -32,6 +32,7 @@ public function tearDown(): void {
 		unset( $this->sut );
 		// Clean up all notifications.
 		global $wpdb;
+		$wpdb->hide_errors(); // Truncate tables can fail if the tables don't exist.
```

I'm a little concerned that this could hide potential errors that would be useful for debugging.

Can we check for table existence before truncating if that's the issue?

---

Yes. I will add that. We usually don't check that a table exists before dropping it or truncating and the error is logged and nothing more happens, but I agree that it'd be best to be a bit stricter here.

---

Also concerned about `$wpdb->hide_errors()`, are there any pointers on why those tables are missing?

---

I've removed the `hide_errors()` call from those tests. What will be, will be.